lib/ethpy/ethpy/hyperdrive/interface/_contract_calls.py

```diff
@@ -188,6 +188,14 @@ async def _async_open_long(
             read_retry_count=interface.read_retry_count,
         )
     if slippage_tolerance is not None:
+        preview_result = smart_contract_preview_transaction(
```

Comment:
this shouldn't be here.

`if slippage_tolerance is not None` will now result in two preview calls. You'll get your preview call from the last conditional.